@vates/read-chunk/index.js

```diff
@@ -65,3 +65,18 @@ exports.readChunkStrict = async function readChunkStrict(stream, size) {
 
   return chunk
 }
+exports.readChunkStrict = readChunkStrict
+
+exports.skipChunk = async function skip(stream, skipLength, chunkLength = 1024 * 1024) {
+  if (skipLength <= 0) {
+    return
+  }
+  let remaining = skipLength
+  while (remaining > chunkLength) {
+    await readChunkStrict(stream, chunkLength)
+    remaining -= chunkLength
+  }
+  if (remaining > 0) {
+    await readChunkStrict(stream, remaining)
+  }
+}
```

Comment:
Different (untested) implementation which does not allocate unnecessary buffers:

```js
async function skip(stream, size) {
  return size === 0 || stream.closed || stream.readableEnded
    ? Promise.resolve(0)
    : new Promise((resolve, reject) => {
        let left = size
        function onEnd() {
          resolve(left - size)
          removeListeners()
        }
        function onError(error) {
          reject(error)
          removeListeners()
        }
        function onReadable() {
          const data = stream.read()
          left -= data.length
          if (left > 0) {
            // continue to read
          } else {
            // if more than wanted has been read, push back the rest
            if (left < 0) {
              stream.unshift(data.slice(left))
            }

            resolve(size)
            removeListeners()
          }
        }
        function removeListeners() {
          stream.removeListener('end', onEnd)
          stream.removeListener('error', onError)
          stream.removeListener('readable', onReadable)
        }
        stream.on('end', onEnd)
        stream.on('error', onError)
        stream.on('readable', onReadable)
        onReadable()
      })
}
exports.skip = skip

exports.skipStrict = async function skipStrict(stream, size) {
  const bytesSkipped = await skip(stream, size)
  if (bytesSkipped !== size) {
    const error = new Error('stream has ended with not enough data')
    error.bytesSkipped = bytesSkipped
    throw error
  }
}
```

Reply:
What about this comment?